evaluator/src/main/kotlin/DependencyRule.kt

```diff
@@ -34,8 +33,7 @@ import org.ossreviewtoolkit.utils.common.enumSetOf
 class DependencyRule(
     ruleSet: RuleSet,
     name: String,
-    pkg: Package,
-    curations: List<PackageCurationResult>,
+    pkg: CuratedPackage,
```

Comment:
- I believe that consuming curations in the rules is an edge case. So, I believe there is hardly a rule interrested in the
  curations
- This change makes rules depend on curated package, so it may become harder to change. There is already one potential future change known [1]

I'm not saying this change shouldn't be done. The argumentation that simplifying passing the parameter to the constructor satisfies the change seems not convincing to me, because there is only few constructor calls, but many places consuming the package (which get a bit more verbose).

Anyhow, would it make sense to first implement [1]?

[1] https://github.com/oss-review-toolkit/ort/issues/5637

Reply:
> I believe that consuming curations in the rules is an edge case. So, I believe there is hardly a rule interrested in the curations

Probably. But passing the whole `CuratedPackage` will become necessary anyway due to the upcoming changes in order to access the concluded license in rules.

Reply:
> Anyhow, would it make sense to first implement #5637?

Doesn't this PR also implicitly resolve that issue (at least for the mentioned `concludedLicense` field) as it moves out that field from the `Package` class?

Reply:
> Doesn't this PR also implicitly resolve that issue (at least for the mentioned `concludedLicense` field) as it moves out that field from the `Package` class?

Yes, it implicitly solves that just for concluded license - but my point was about introducing dependencies from rules to a data model which we already know would change. For that, it doesn't matter whether the problem is partially solved for the concluded license IMO.